dataproc_jupyter_plugin/dagTemplates/pysparkJobTemplate-v1.txt

```diff
@@ -0,0 +1,155 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta, timezone
+from airflow import DAG
+from airflow.providers.google.cloud.operators.dataproc import DataprocSubmitJobOperator
+from airflow.operators.python_operator import PythonOperator
+from google.cloud import dataproc_v1
+from google.api_core.client_options import ClientOptions
+
+default_args = {
+    'owner': '{{owner}}',
+    'start_date': '{{start_date}}',
+    'retries': '{{retry_count}}',
+    'retry_delay': timedelta(minutes=int('{{retry_delay}}')), 
+    'email': {{email}}, 
+    'email_on_failure': '{{email_failure}}',     
+    'email_on_retry': '{{email_delay}}',     
+    'email_on_success':'{{email_success}}'
+}
+time_zone = '{{time_zone}}'
+stop_cluster_check = '{{stop_cluster}}'
+input_notebook = '{{input_notebook}}'
+output_notebook = '{{output_notebook}}'
+parameters = '''
+{{parameters}}
+'''
+notebook_args= [input_notebook, output_notebook] 
+# Check if parameters is not empty or contains only whitespace
+if parameters.strip():  
+    notebook_args.extend(["--parameters", parameters])
+
+def get_client_cert():
+    # code to load client certificate and private key.
+    return client_cert_bytes, client_private_key_bytes
+ 
+
+def get_cluster_state_start_if_not_running():
+
+    options = ClientOptions(api_endpoint="{{gcpRegion}}-dataproc.googleapis.com:443",
+    client_cert_source=get_client_cert)
+
+    # Create a client
+    client = dataproc_v1.ClusterControllerClient(client_options=options)
+
+    # Initialize request argument(s)
+    request = dataproc_v1.GetClusterRequest(
+        project_id='{{gcpProjectId}}',
+        region='{{gcpRegion}}',
+        cluster_name='{{cluster_name}}',
+    )
+
+    # Make the request
+    response = client.get_cluster(request=request)
+    
+   
+    # Handle the response
+    print(f"State is {response.status.state}")
+    if response.status.state in (6, 7):
+        print("Cluster is in stopped/stopping state. Starting the cluster")
+        request1 = dataproc_v1.StartClusterRequest(
+        project_id='{{gcpProjectId}}',
```

Comment:
The indentation of this line and the two following it are wrong.

Reply:
Resolved - Formatted indentation.